cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1243,31 +1249,56 @@ if (ARROW_FLIGHT)
       ${EP_LOG_OPTIONS}
       CMAKE_ARGS ${GRPC_CMAKE_ARGS}
       ${EP_LOG_OPTIONS})
-    include_directories(SYSTEM ${GRPC_INCLUDE_DIR})
+
+    set(GPR_STATIC_LIB "${GRPC_STATIC_LIBRARY_GPR}")
+    set(GRPC_STATIC_LIB "${GRPC_STATIC_LIBRARY_GRPC}")
+    set(GRPCPP_STATIC_LIB "${GRPC_STATIC_LIBRARY_GRPCPP}")
+    set(GRPC_ADDRESS_SORTING_STATIC_LIB "${GRPC_STATIC_LIBRARY_ADDRESS_SORTING}")
+    # XXX(wesm): relying on vendored c-ares provided by gRPC for the time being
+    set(CARES_STATIC_LIB "${GRPC_STATIC_LIBRARY_CARES}")
+    set(GRPC_CPP_PLUGIN "${GRPC_BUILD_DIR}/${CMAKE_CFG_INTDIR}/grpc_cpp_plugin")
+    # get_property(GPR_STATIC_LIB TARGET gRPC::gpr PROPERTY LOCATION)
+    # get_property(GRPC_STATIC_LIB TARGET gRPC::grpc_unsecure PROPERTY LOCATION)
+    # get_property(GRPCPP_STATIC_LIB TARGET gRPC::grpc++_unsecure PROPERTY LOCATION)
+    # get_property(GRPC_ADDRESS_SORTING_STATIC_LIB
+    #   TARGET gRPC::address_sorting PROPERTY LOCATION)
+    # # XXX(wesm): relying on vendored c-ares provided by gRPC for the time being
+    # get_property(CARES_STATIC_LIB TARGET c-ares::cares_static PROPERTY LOCATION)
+    # # Get location of grpc_cpp_plugin so we can pass it to protoc
+    # get_property(GRPC_CPP_PLUGIN TARGET gRPC::grpc_cpp_plugin PROPERTY LOCATION)
   else()
-    find_package(gRPC CONFIG REQUIRED)
+    find_package(gRPC REQUIRED)
     set(GRPC_VENDORED 0)
   endif()
 
-  get_property(GPR_STATIC_LIB TARGET gRPC::gpr PROPERTY LOCATION)
+  # If we built gRPC ourselves, we should use its c-ares.
+  if ("${CARES_STATIC_LIB}" STREQUAL "")
```

Comment:
I do not see `CARES_STATIC_LIB` used anywhere in our build files. Am I missing something?

Reply:
It's used below in `ADD_THIRDPARTY_LIB` - it's not actually used by Arrow/Flight, but by gRPC.

Reply:
Ah, sorry :-)